docs/reference/cluster/stats.asciidoc

```diff
@@ -338,7 +348,289 @@ This object is not populated by the cluster stats API.
 +
 To get information on segment files, use the <<cluster-nodes-stats,node stats
 API>>.
+====
+
+[[cluster-stats-api-response-body-nodes]]
+===== `nodes` section
+
+[%collapsible]
+====
+`nodes.count.total`::
+(integer)
+Total number of nodes selected by the request's <<cluster-nodes,node filters>>.
+
+`nodes.count.coordinating_only`::
+(integer)
+Number of selected <<coordinating-only-node,coordinating only>> nodes.
+
+`nodes.count.data`::
+(integer)
+Number of selected nodes with the <<data-node,data>> role.
+
+`nodes.count.ingest`::
+(integer)
+Number of selected nodes with the <<node-ingest-node,ingest>> role.
+
+`nodes.count.master`::
+(integer)
+Number of selected nodes with the <<master-node,master-eligible>> role.
+
+`nodes.count.ml`::
+(integer)
+Number of selected nodes with the <<ml-node,machine learning>> role.
+
+`nodes.count.voting_only`::
+(integer)
+Number of selected nodes with the <<voting-only-node,voting-only>> role.
```

Comment:
Should we collapse these into one entry? The `nodes.count.coordinating_node` field is special in that it's for any nodes the have no role and will always be a valid entry, but roles are pluggable now so other values could end up here over time. 

If that feels like a bad idea from a docs perspective, then these are fine as-is.

Reply:
This is a good idea. I left `coordinating_only` as-is but I used `nodes.count.<role>` for the remaining parms. Changes are in 5876819e1309b5d498a86fcfe2bd099be4aefc92.